upgrade/upgrade-provider.go

```diff
@@ -303,7 +307,39 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 		steps = append(steps, step.Cmd(exec.CommandContext(ctx,
 			"go", "get", "github.com/pulumi/pulumi-terraform-bridge/v3@"+targetBridgeVersion)).
 			In(repo.providerDir()))
+		// Now that we've updated the bridge version, we need to update the corresponding pulumi version in sdk/go.mod.
+		// It needs to match the version used in provider/go.mod, which is *not* necessarily `latest`.
+		var newSdkVersion string
+		getNewPulumiVersionStep := step.F("Get Pulumi SDK version", func() (string, error) {
+			path := repo.root
+			file := filepath.Join(path, "provider", "go.mod")
+
+			modFile, err := os.Open(file)
+
+			scanner := bufio.NewScanner(modFile)
+			var pulumiImportLine []string
+			for scanner.Scan() {
```

Comment:
Instead of a line-by-line scan, can we just parse the file?

This is what we do in other places to get a version:
https://github.com/pulumi/upgrade-provider/blob/05e15264acfbeab0135abc0ec6ecfaeea0b3da34/upgrade/steps-helpers.go#L56-L61


---

We already have a function that is very similar to this code: `originalGoVersionOf`. Could we refactor this lookup to `currentGoVersionOf` and share some logic?